aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -511,6 +518,23 @@ module aptos_framework::stake {
         let amount = coin::value<AptosCoin>(&coins);
         assert!(amount > 0, error::invalid_argument(EINVALID_STAKE_AMOUNT));
 
+        // Only track and validate voting power increase for active and pending_active validator.
+        // Pending_inactive validator will be removed from the validator set in the next epoch.
+        // Inactive validator's total stake will be tracked when they join the validator set.
+        let validator_state = get_validator_state(pool_address);
```

Comment:
btw I think we can do the lookup more efficiently than loop over the entire 3 vectors. if it's active, we can use the validator index directly from the pool address

Reply:
Good idea! Updated.